### PR TITLE
Properly handle @keyframes

### PR DESCRIPTION
**Pull request by Schnouki**
While using scope-css I encountered an issue with a stylesheet that used `@keyframes animation-name {}` to create animations. So I fixed it :wink: 

This is a little complex but should be reasonably robust.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
 module.exports = scope;
 scope.replace = replace;
 
-function scope (css, parent) {
+function scope (css, parent, prefix) {
 	if (!css) return css;
 
 	if (!parent) return css;
 
+	if (!prefix) prefix = 'p' + Math.random().toString(36).substring(7) + '-';
```
Comment (dy) [inline index.js]: love the solution

**Comment by dy** [comment]
Do I understand right that you want to make animations specific to a parent selector?
That seems to be an issue of of the package's scope.

**Comment by Schnouki** [comment]
Okay, I probably need to explain why I did that :)

I'm using this package in a TamperMonkey / GreaseMonkey userscript, to add some features to an existing website (to be more precise: the admin section of a e-commerce website hosted on Shopify). I inject some JS libraries (from unpkg.com), and one of them needs a CSS (also from unpkg.com) that includes an animation. To do that correctly, I need to use `scope-css`, to make sure that this CSS only affects the few `<div>`s I add to the page, and not the rest of that page.

The problem is that this CSS becomes invalid after processing it with `scope-css`: it turns

    @keyframes loading {
        from { top: 0; }
        to { top: 100px; }
    }

into

    @keyframes loading {
        .parent from { top: 0; }
        .parent to { top: 100px; }
    }

which is not valid syntax.

The obvious solution is, of course, to patch `scope-csss` to just remove the `.parent` prefix from the animation keyframes... but then I have another issue: the webpage already defines an animation named `loading`...

Since I started using `scope-css` exactly to prevent such issues which conflicting CSS rules, it made sense for me to patch it to rename animations so that they become specific to a scope. 

But if you think this is far-fetched and shouldn't be part of `scope-css`, no problem, I understand :)

**Comment by dy** [comment]
Ok, keyframes are at least broken.
I don't like randomish nature of prefix generation, as well as custom prefix argument at all. Would that be enough to slugify parent selector for that purpose? Eg.

```css
@keyframes parent-selector-loading {
    from { top: 0; }
    to { top: 100px; }
}
```

**Comment by dy** [comment]
Although... in case if a css is just a part of another css, which needs to be scoped, then prefixing animations would break other concatenated css parts. I think that is a matter of a separate module though, something like `prefix-keyframes`.
Although I appreciate your efforts, I will merge part fixing keyframes, thanks @Schnouki

**Comment by dy** [comment]
Ok, added in 1.2.0

**Pull request closed without merging**